src/transformers/trainer.py

```diff
@@ -1076,7 +1077,7 @@ def _wrap_model(self, model, training=True):
                 ).to(self.args.device)
 
         elif is_sagemaker_dp_enabled():
-            model = DDP(model, device_ids=[dist.get_local_rank()], broadcast_buffers=False)
+            model = DDP(model, device_ids=[int(os.getenv("SMDATAPARALLEL_LOCAL_RANK"))])
```

Comment:
```suggestion
            model = nn.parallel.DistributedDataParallel(model, device_ids=[int(os.getenv("SMDATAPARALLEL_LOCAL_RANK"))])
```
Let's be consistent with the torch distributed version.